tick/stateful/scope.go

```diff
@@ -19,14 +31,16 @@ var empty = new(interface{})
 type Scope struct {
 	variables map[string]interface{}
 
-	dynamicMethods map[string]*DynamicMethod
+	dynamicMethods map[string]DynamicMethod
+	dynamicFuncs   map[string]*DynamicFunc
```

Comment:
So that function signature of 
```
func (s *Scope) DynamicFunc(name string) *DynamicFunc {
	return s.dynamicFuncs[name]
}
```
doesn't have to be
```
func (s *Scope) DynamicFunc(name string) (*DynamicFunc, bool)
```
https://github.com/influxdata/kapacitor/blob/md-type-system/tick/stateful/scope.go#L90

So we can do a nil check in [Type] and [callFunction](https://github.com/influxdata/kapacitor/blob/md-type-system/tick/stateful/eval_function_node.go#L87).

Happy to change it to return the `bool` if you think thats better.